lib/plugin.rb

```diff
@@ -21,18 +23,32 @@ def update
   private
 
   def zapp_plugin
-    get_request(plugins_url, request_params)
+    plugin_candidates = get_request(plugins_url, request_params)
       .body
-      .find { |p| p["name"] == @name || p["external_identifier"] == @identifier }
+      .select do |p|
+        p["name"] == @name || identifier_matches?(p)
+      end
+
+    return nil if plugin_candidates.count == 0
+    return plugin_candidates.first if plugin_candidates.count == 1
```

Comment:
why we need this> isn't mapping solving the case of 1? is it an optimization?

Reply:
Well, the idea is that if the request finds one plugin, I use this one, otherwise I add a prompt for selection. In one case I return a plugin, and in he other, I return an array. But maybe I can handle the three cases in a switch ?